blockchain/chain.go

```diff
@@ -1304,15 +1304,6 @@ type Config struct {
 	// time is adjusted to be in agreement with other peers.
 	TimeSource MedianTimeSource
 
-	// Notifications defines a callback to which notifications will be sent
-	// when various events take place.  See the documentation for
-	// Notification and NotificationType for details on the types and
-	// contents of notifications.
-	//
-	// This field can be nil if the caller is not interested in receiving
-	// notifications.
-	Notifications NotificationCallback
```

Comment:
I would much rather this be left as an option in the config struct and changed to an array.  That allows it to be treated as immutable and thus safe for concurrent access without the need to introduce a mutex.

The changes in this PR, as they are, are not safe for concurrent access since it would be possible to subscribe new listeners at any time and there is no mutex to protect it.  Not only would you need the mutex on the subscribe side, but you'd have to take it every time you want to range the `s.notifications` too.